typescript/infra/src/core/deploy.ts

```diff
@@ -47,38 +54,10 @@ export class HyperlaneCoreInfraDeployer<
       const startingBlockNumber = this.startingBlockNumbers[chain];
 
       if (startingBlockNumber) {
-        rustConfig.index = { from: startingBlockNumber.toString() };
+        chainConfig.index = { from: startingBlockNumber.toString() };
       }
-
-      this.multiProvider
-        .remoteChains(chain)
-        .filter((_) => configChains.includes(_))
-        .forEach((remote) => {
-          // The agent configuration file should contain the `chain`'s inbox on
-          // all the remote chains
-          const remoteContracts = this.deployedContracts[remote];
-          const inboxContracts =
-            remoteContracts?.inboxes?.[chain as Exclude<Chain, Chain>];
-
-          const metadata = chainMetadata[remote];
-          const inbox = {
-            domain: metadata.id.toString(),
-            name: remote,
-            rpcStyle: 'ethereum',
-            finalityBlocks: metadata.finalityBlocks.toString(),
-            connection: {
-              type: ConnectionType.Http,
-              url: '',
-            },
-            addresses: {
-              inbox: inboxContracts?.inbox.address,
-              validatorManager: inboxContracts?.inboxValidatorManager.address,
-            },
-          } as const;
-
-          rustConfig.inboxes[remote] = inbox;
-        });
-      writeJSON(directory, `${chain}_config.json`, rustConfig);
+      rustConfig.chains[chain] = chainConfig;
     });
+    writeJSON(directory, 'rust_config.json', rustConfig);
```

Comment:
I take it the accompanying v2 agent PR requires a config like this? (I haven't taken a look yet)

Reply:
It does indeed